### DISCUSSION TITLE
`IMask.AnyMaskedOptions` type missing

### DISCUSSION THREAD

**Discussion by bombillazo** [uNmAnNeR/imaskjs#930] [Q&A]
**Describe the bug**
I am looking for the type `IMask.AnyMaskedOptions` as used in different examples and libraries but I cannot find it in the latest version. Seems like it was removed in v7.0.0.

**To Reproduce**
N/A

**Expected behavior**
Somehow get the type equivalent to `IMask.AnyMaskedOptions` that was previously implemented (at least up til version 6.6.3)

**Environment:**
 - IMask version > v7.0.0

**Additional context**

**Comment by npearson72** [comment]
@uNmAnNeR any word on this?

**Comment by AlexKorole** [comment]
Same problem: Namespace 'IMask' has no exported member 'AnyMaskedOptions'
```
mask?: IMask.AnyMaskedOptions
```

**Comment by uNmAnNeR** [comment]
yes it was removed. Now you can use `FactoryArg` or `FactoryOpts` instead
check the API for available types:
https://imask.js.org/api/modules/imask.html

**Comment by hassanasad** [comment]
@uNmAnNeR Would it be possible for you to give an example of how to use the `FactoryArg` or `FactoryOpts` types. I can't seem to find an example in the documentation about it. (I am using Angular package for the imask `v7.2.0`)

My test code:
```
import IMask from 'imask';

export const MonthYearMask: any = {
	mask     : 'MM / YYYY',
	overwrite: true,
	autofix  : true,
	lazy     : true,
	blocks   : {
		MM: { mask: IMask.MaskedRange, placeholderChar: 'M', from: 1, to: 12, maxLength: 2 },
		YYYY: { mask: IMask.MaskedRange, placeholderChar: 'Y', from: 1800, to: 9999, maxLength: 4 },
	},
};

export const FreeDigitMask: any = {
	mask  : /^\d+$/,
	signed: false,
	scale : 0,
};
```

**Reply by bombillazo** [reply]
Yes, an example would be ideal to make migrating between versions much easier.

**Reply by hassanasad** [reply]
Figured it out:

```
import IMask, { FactoryArg } from 'imask';

export const MonthYearMask: FactoryArg = {
	mask     : 'MM / YYYY',
	overwrite: true,
	autofix  : true,
	lazy     : true,
	blocks   : {
		MM: { mask: IMask.MaskedRange, placeholderChar: 'M', from: 1, to: 12, maxLength: 2 },
		YYYY: { mask: IMask.MaskedRange, placeholderChar: 'Y', from: 1800, to: 9999, maxLength: 4 },
	},
};

export const FreeDigitMask: FactoryArg = {
	mask  : /^\d+$/,
	min   : 0,
	scale : 0,
};
```

`signed` property has been replaced with `min` and type `FactoryArg` added.

Hope it helps.

**Reply by uNmAnNeR** [reply]
@hassanasad if it's only for pattern mask then you can narrow type to `MaskedPatternOptions`.
Not sure if `FreeDigitMask` is correct. There are no `scale` and `min` options for `RegExp` mask or is it a custom one?

**Reply by hassanasad** [reply]
@uNmAnNeR Thanks, i will have a look at it. These masks were written few years ago and may need to be refactored.

**Comment by mertafor** [comment]
These type definitions are not available for react-imask library. I was able to access type definitions (E.g. FactoryArg) by installing imask as a devdependency, which doesn't look like an ideal solution. Is it possible to have access these type definitions from react-imask as well?

**Comment by baluram-kumavat-acute** [comment]
@uNmAnNeR I am using like this 

import Imask from "imask";

export type IMaskInputProps = Partial<
  Pick<
    Imask.AnyMaskedOptions,
    | "mask"
    | "value"
    | "prepare"
    | "validate"
    | "commit"
    | "overwrite"
    | "placeholderChar"
    | "lazy"
    | "definitions"
    | "blocks"
    | "pattern"
    | "format"
    | "parse"
    | "autofix"
    | "radix"
    | "thousandsSeparator"
    | "mapToRadix"
    | "scale"
    | "signed"
    | "normalizeZeros"
    | "padFractionalZeros"
    | "min"
    | "max"
    | "dispatch"
  >
> & {
  unmask?: boolean;
  onAccept?: <T>(...args: T[]) => void;
  onComplete?: <T>(...args: T[]) => void;
};






----------------------------------------------------------------------------
second code where is used above props (IMaskInputProps )

interface extendedProps {
  MaskProps: IMaskInputProps;
  formattedValue?: boolean;
}

export type AllInputMaskProps = Merge<TextFieldProps, extendedProps>;

const MyInputMaskCustom: FC<AllInputMaskProps> = ({
  MaskProps,
  formattedValue,
  ...others
}) => {
  return (
    <TextField
      {...others}
      InputProps={{
        inputComponent: InputMaskCustom,
        inputProps: { MaskProps: { ...MaskProps, formattedValue } },
      }}
    />
  );
}; 

  now how can i add  FactoryArg or FactoryOpts ?


![Screenshot 2024-05-27 151635](https://github.com/uNmAnNeR/imaskjs/assets/141899347/6de5eaf6-42a5-416a-8278-4c8c278a9dd0)
![Screenshot 2024-05-27 151727](https://github.com/uNmAnNeR/imaskjs/assets/141899347/0cf27728-5b2f-4ed4-b201-ae1f1540ed97)
